src/main/java/me/senseiwells/essentialclient/clientscript/values/PlayerValue.java

```diff
@@ -838,6 +841,77 @@ private Value craftRecipe(Arguments arguments) throws CodeError {
 			return NullValue.NULL;
 		}
 
+		@FunctionDoc(
+			name = "craftRecipe",
+			desc = "This allows you to craft a predefined recipe",
+			params = {
+				RECIPE, "recipe", "the recipe you want to craft",
+				BOOLEAN, "boolean", "whether result should be dropped or not"
+			},
+			throwMsgs = "Must be in a crafting GUI",
+			example = "player.craftRecipe(Recipe.CHEST, true);"
+		)
+		private Value craftRecipeDrop(Arguments arguments) throws CodeError {
+			MinecraftClient client = ArucasMinecraftExtension.getClient();
+			if (!(client.currentScreen instanceof HandledScreen<?> handledScreen)) {
+				throw arguments.getError("Must be in a crafting GUI");
+			}
+			ClientPlayerInteractionManager interactionManager = ArucasMinecraftExtension.getInteractionManager();
+			RecipeValue recipeValue = arguments.skip().getNext(RecipeValue.class);
+			BooleanValue booleanValue = arguments.getNext(BooleanValue.class);
+			client.execute(() -> {
+				CraftingSharedConstants.IS_SCRIPT_CLICK.set(true);
+				interactionManager.clickRecipe(handledScreen.getScreenHandler().syncId, recipeValue.value, true);
+				if (booleanValue.value) {
+					InventoryUtils.shiftClickSlot(client, handledScreen, 0);
+				}
+				else {
+					InventoryUtils.dropStackScheduled(client, handledScreen, true);
+				}
+			});
+			return NullValue.NULL;
+		}
+
+		@FunctionDoc(
+			name = "clickRecipe",
+			desc = "This allows you to click a predefined recipe",
+			params = {RECIPE, "recipe", "the recipe you want to select"},
+			throwMsgs = "Must be in a crafting GUI",
+			example = "player.clickRecipe(Recipe.CHEST);"
+		)
+		private Value clickRecipeDefault(Arguments arguments) throws CodeError {
+			MinecraftClient client = ArucasMinecraftExtension.getClient();
+			if (!(client.currentScreen instanceof HandledScreen<?> handledScreen)) {
+				throw arguments.getError("Must be in a crafting GUI");
+			}
+			ClientPlayerInteractionManager interactionManager = ArucasMinecraftExtension.getInteractionManager();
+			RecipeValue recipeValue = arguments.skip().getNext(RecipeValue.class);
+			client.execute(() -> interactionManager.clickRecipe(handledScreen.getScreenHandler().syncId, recipeValue.value, false));
```

Comment:
I'm sorry I was wrong, here you do need `IS_SCRIPT_CLICK.set(true)` in order for it to function correctly since the client isn't guarenteed to have all the recipes unlocked. The game goes for the other `clickRecipe` method